src/plugins/ImportArtifact/ImportArtifact.js

```diff
@@ -66,23 +66,28 @@ define([
         const parent = await this.getArtifactsDir();
         const dataNode = this.core.createNode({base, parent});
 
-        const name = await this.getAssetName(hash) ||
+        const name = await this.getAssetName(hashOrPath) ||
             baseName[0].toLowerCase() + baseName.substring(1);
-
-        const assetInfo = await this.transfer(hash, config.storage, name);
-        this.core.setAttribute(dataNode, 'data', JSON.stringify(assetInfo));
-        this.core.setAttribute(dataNode, 'type', baseName);
-        this.core.setAttribute(dataNode, 'createdAt', Date.now());
+        let assetInfo;
 
         try {
+            if(config.dataHash){
```

Comment:
This is really doing two completely different things and it might be nice to make that clear from the beginning. Maybe have a variable for `const uploadNewArtifact = !!config.dataHash;`. 

However, this is a little problematic because it would be hard to give a good error message if nothing is provided (it is not clear if the hash was omitted or the path). That said, it might be cleaner to actually just make two different plugins: `UploadArtifact` and `ImportArtifact`. This would make the code a bit cleaner. It appears the only shared logic is:
- ensuring that `Data` exists in the metamodel
- `getArtifactsDir`
- setting the attributes for the newly created artifact (pretty minor - duplication is probably fine)